### ISSUE TITLE
potential jax 0.4.35 release issue?

### ISSUE THREAD

**Issue by haohuanw** [jax-ml/jax#24826]
### Description

i am checking jax 0.4.35's pip wheel metadata and found that `cuda12-pip` variant depends on `jaxlib==0.4.34` is this expected?

```
Metadata-Version: 2.1
Name: jax
Version: 0.4.35
Summary: Differentiate, compile, and transform Numpy code.
Home-page: https://github.com/jax-ml/jax
Author: JAX team
Author-email: jax-dev@google.com
License: Apache-2.0
Classifier: Programming Language :: Python :: 3.10
Classifier: Programming Language :: Python :: 3.11
Classifier: Programming Language :: Python :: 3.12
Requires-Python: >=3.10
Description-Content-Type: text/markdown
License-File: LICENSE
License-File: AUTHORS
Requires-Dist: jaxlib<=0.4.35,>=0.4.34
Requires-Dist: ml-dtypes>=0.4.0
Requires-Dist: numpy>=1.24
Requires-Dist: opt-einsum
Requires-Dist: scipy>=1.10
Requires-Dist: numpy>=1.26.0; python_version >= "3.12"
Requires-Dist: scipy>=1.11.1; python_version >= "3.12"
Provides-Extra: ci
Requires-Dist: jaxlib==0.4.34; extra == "ci"
Provides-Extra: cpu
Provides-Extra: cuda
Requires-Dist: jaxlib==0.4.34; extra == "cuda"
Requires-Dist: jax-cuda12-plugin[with_cuda]<=0.4.35,>=0.4.34; extra == "cuda"
Provides-Extra: cuda12
Requires-Dist: jaxlib==0.4.34; extra == "cuda12"
Requires-Dist: jax-cuda12-plugin[with_cuda]<=0.4.35,>=0.4.34; extra == "cuda12"
Provides-Extra: cuda12_local
Requires-Dist: jaxlib==0.4.34; extra == "cuda12-local"
Requires-Dist: jax-cuda12-plugin==0.4.34; extra == "cuda12-local"
Provides-Extra: cuda12_pip
Requires-Dist: jaxlib==0.4.34; extra == "cuda12-pip"   **<= section in question**
Requires-Dist: jax-cuda12-plugin[with_cuda]<=0.4.35,>=0.4.34; extra == "cuda12-pip"
Provides-Extra: k8s
Requires-Dist: kubernetes; extra == "k8s"
Provides-Extra: minimum-jaxlib
Requires-Dist: jaxlib==0.4.34; extra == "minimum-jaxlib"
Provides-Extra: tpu
Requires-Dist: jaxlib<=0.4.35,>=0.4.34; extra == "tpu"
Requires-Dist: libtpu-nightly==0.1.dev20241010+nightly.cleanup; extra == "tpu"
Requires-Dist: libtpu==0.0.2; extra == "tpu"
Requires-Dist: requests; extra == "tpu"
```

### System info (python version, jaxlib version, accelerator, etc.)

N/A

**Comment by mattjj** [comment]
Thanks for raising this! Sounds like it's a known issue and we're going to fix it in the next release.

**Comment by traversaro** [comment]
I found a inconsistency between jax 0.4.35 sdist available on PyPI and the source code available at a tag `jax-v0.4.35` . By inspecting the values of the `_minimum_jaxlib_version` and `_libtpu_version ` variables, it seems that the sdist at https://files.pythonhosted.org/packages/e3/34/21da583b9596e72bb8e95b6197dee0a44b96b9ea2c147fccabd43ca5515b/jax-0.4.35.tar.gz contains the source code at the commit https://github.com/jax-ml/jax/commit/e4f3f8f064228c0739b3333a9e78ac715cb34273, while the tag https://github.com/jax-ml/jax/blob/jax-v0.4.35 points to https://github.com/jax-ml/jax/commit/81991d87c8608d07f9752940375092aa4fb9610e .

**Comment by hawkinsp** [comment]
Yes, it's a known issue. JAX 0.4.35's sdist and `jax` wheel were built at a slightly incorrect commit.

It is fixed for v0.4.36, but we cannot fix it retroactively for v0.4.35, and it doesn't really merit yanking anything because everything should still work fine.

**Comment by hawkinsp** [comment]
(Note: we already released v0.4.36)

**Comment by traversaro** [comment]
> Yes, it's a known issue. JAX 0.4.35's sdist and `jax` wheel were built at a slightly incorrect commit.
> 
> It is fixed for v0.4.36, but we cannot fix it retroactively for v0.4.35, and it doesn't really merit yanking anything because everything should still work fine.

Thanks for the clarification! Yes, everything is working fine for 0.4.35 so there is no need for yanking (and sorry I gave the impression to suggest that). I just wrote this down in case someone else was confused by this (as I was).
